dotCMS/src/integration-test/java/com/dotcms/content/elasticsearch/business/ESContentFactoryImplTest.java

```diff
@@ -363,4 +387,62 @@ public void testDeleteVersion_KeepPermissions() throws DotSecurityException, Dot
         }
     }
 
+    @Test
+    @UseDataProvider("testCases")
+    public void testIndexSearchFilteringByDates(final TestCase testCase){
+        final long languageId = APILocator.getLanguageAPI().getDefaultLanguage().getId();
+        final ContentType contentType = TestDataUtils.getNewsLikeContentType();
+
+        LocalDateTime today, yesterday, tomorrow;
+
+        final LocalDateTime now = LocalDateTime.now();
+
+        if (testCase.formatPattern.equals("hh:mm:ssa") || testCase.formatPattern.equals("hh:mm:ss a") ||
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Position literals first in String comparisons](https://app.codacy.com/app/dotCMS/core/pullRequest?prid=4545506)